cocos2d/core/platform/CCTypes.js

```diff
@@ -37,7 +37,7 @@ cc.Color = function (r, g, b, a) {
     r = r || 0;
     g = g || 0;
     b = b || 0;
-    a = a || 0;
+    a = (a == null) ? 255 : a;
```

Comment:
a == null is not strict enough, for example `0 == null` is falsy then user will get 255 as alpha.

Please use `isNaN(a) ? 255 : a`